### ISSUE TITLE
Test coverage (init, join, upgrade) for matrix of GA supported options

### ISSUE THREAD

**Issue by neolit123** [kubernetes/kubeadm#968]
1.12 planning doc:
https://docs.google.com/document/d/1YP8sCtgo58yqQ9sR4nsn9iuVFmglCCBMivCTwjd1ERE/edit#

items to support:
- [ ] External etcd, internal etcd
- [ ] Single master
- [ ] Static pods
- [ ] dynamic kubelet config?
- [ ] Remove kubernetes anywhere

**Comment by timothysc** [comment]
@rdodev  this is still all manual and we should help make this automated now.

**Comment by neolit123** [comment]
adding some up to date comments:
> - [ ] External etcd, internal etcd

we don't test external etcd yet.

> - [ ] Single master

i don't think we have to have separate tests with workloads on a single master.
this will not run conformance too.

> - [ ] Static pods

our CP components already run in static pods.
additional static pod tests seems like something SIG node should test, IMO.

> - [ ] dynamic kubelet config?

this feature is slowly being phased out of support.

> - [ ] Remove kubernetes anywhere

soon.

**Comment by rdodev** [comment]
I'm no longer allocated in this line of work, so other willing soul will have to step forward and drive this.

**Comment by neolit123** [comment]
the ticket is out of date so i'm going to close it.

> External etcd, internal etcd

external etcd is already being tested. this is tracked in the HA to beta support ticket.
internal etcd is tested by default.

>  Single master

single master can be tested with the conformance suite but only in parallel mode. serial mode would fail. given this complication i would prefer if we don't have a test that has workloads on a single cp node.
let's open a new issue if this is desired. given latest kind moved containerd we were discussing having single-cp-node tests that test docker, but Fabrizio mentioned that we can have kinder support multiple CRs.

> Static pods

we already doing that.

>  dynamic kubelet config?

the issue is phased out.

>  Remove kubernetes anywhere

the repo is still there but we are no longer using it.
